### ISSUE TITLE
Rancher deployed on k3s with traefik and external TLS doesn't respond to /ping properly

### ISSUE THREAD

**Issue by dnoland1** [rancher/rancher#26886]
**Version:**
```
# k3s -v
k3s version v1.17.4+k3s1 (3eee8ac3)
```

**K3s arguments:**
usr/local/bin/k3s server --cluster-cidr 172.16.0.0/16 --service-cidr 192.168.0.0/16 --cluster-dns 192.168.0.10 --flannel-backend=ipsec

**Describe the bug**
When Rancher is running on k3s cluster with traefik and set to use a load balancer with external TLS, sending a /ping request returns `OK` and not `pong`:
```
$ curl -v https://myhost/ping
*   Trying W.X.Y.Z...
* TCP_NODELAY set
* Connected to myhost (W.X.Y.Z) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*   CAfile: /etc/ssl/cert.pem
  CApath: none
* TLSv1.2 (OUT), TLS handshake, Client hello (1):
* TLSv1.2 (IN), TLS handshake, Server hello (2):
* TLSv1.2 (IN), TLS handshake, Certificate (11):
* TLSv1.2 (IN), TLS handshake, Server key exchange (12):
* TLSv1.2 (IN), TLS handshake, Server finished (14):
* TLSv1.2 (OUT), TLS handshake, Client key exchange (16):
* TLSv1.2 (OUT), TLS change cipher, Change cipher spec (1):
* TLSv1.2 (OUT), TLS handshake, Finished (20):
* TLSv1.2 (IN), TLS change cipher, Change cipher spec (1):
* TLSv1.2 (IN), TLS handshake, Finished (20):
* SSL connection using TLSv1.2 / ECDHE-RSA-AES128-GCM-SHA256
* ALPN, server accepted to use h2
* Server certificate:
*  subject: CN=*.mydomain
*  start date: Apr 14 00:00:00 2020 GMT
*  expire date: May 14 12:00:00 2021 GMT
*  subjectAltName: host "myost" matched cert's "*.mydomain"
*  issuer: C=US; O=Amazon; OU=Server CA 1B; CN=Amazon
*  SSL certificate verify ok.
* Using HTTP2, server supports multi-use
* Connection state changed (HTTP/2 confirmed)
* Copying HTTP/2 data in stream buffer to connection buffer after upgrade: len=0
* Using Stream ID: 1 (easy handle 0x7f879a805400)
> GET /ping HTTP/2
> Host: myhost
> User-Agent: curl/7.64.1
> Accept: */*
>
* Connection state changed (MAX_CONCURRENT_STREAMS == 128)!
< HTTP/2 200
< date: Wed, 15 Apr 2020 00:26:33 GMT
< content-type: text/plain; charset=utf-8
< content-length: 2
<
* Connection #0  to host myhost left intact
OK* Closing connection 0
```
This is problematic because the rancher 2 terraform provider is expecting /ping to return /pong and fails when using the bootstrap resource. See https://github.com/terraform-providers/terraform-provider-rancher2/issues/317

**To Reproduce**
1. set up a k3s cluster, use default option of traefik as the ingress controller. (don't pass --disable traefik). Set this up on AWS.
2. Install Rancher using Helm. Can follow directions on installing Rancher on k3s. Make sure to pass tls = external option.
3. Configure a layer 7 application load balancer (ALB) in AWS. Target group should map to port 80 of the AWS instance running k3s.
4. Curl with /ping 

**Expected behavior**
`/ping` should return pong

**Actual behavior**
`/ping` returns OK . `/ping` is getting intercepted by traefik and not passed through to Rancher.

**Additional context / logs**
Out of box, traefik is configured to respond to `/ping`. See https://docs.traefik.io/operations/ping/#configuration-options This configuration conflicts with Rancher and the ability to use the bootstrap resource in the rancher2 terraform provider.

**Comment by dnoland1** [comment]
@cjellick that seems like a reasonable solution, as long as we can change the rancher2 terraform provider to use this new endpoint.

**Comment by cjellick** [comment]
Yes, that was my thought. Might need to do a little more work to make it backwards compat in the terraform provider. CC @rawmind0

**Comment by rayterrill** [comment]
Just ran into this exact same issue - spent a bunch of time trying to figure out what the heck was going on. Thanks for reporting @dnoland1! Cheers.

**Comment by cjellick** [comment]
We think that with the upgrade to traefik 2.0, the `/ping` endpoint may be off by default. So, this might be resolved in more recent versions of k3s (1.21 and up, I think). We'll have someone check that out.

**Comment by cjellick** [comment]
Marking as internal since this affects hosted rancher.

**Comment by brandond** [comment]
@dnoland1 if you are using external SSL with Traefik v1, and don't want the ping handler operational on the http endpoint, you should do something like this:

```yaml
apiVersion: helm.cattle.io/v1
kind: HelmChartConfig
metadata:
  name: traefik
  namespace: kube-system
spec:
  valuesContent: |-
    ssl:
      enabled: false
    startupArguments:
    - "--ping"
    - "--ping.entrypoint=httpn"
    pingEntryPoint: httpn
```

Note that the Traefik v1 chart requires disabling the ssl endpoint in order to create the alternate plaintext `httpn` endpoint on 8800 that hosts the ping endpoint, which is required for pod healthchecks.

**Comment by cjellick** [comment]
Spoke with @dnoland1. This explanation (plus it not being an issue in traefik 2) is enough for him. However, this could possibly be picked up as an enhancment to the documentation for installing rancher on top of k3s.
